crawlerr/services/rag_service.py

```
@@ -73,6 +128,228 @@ def chunk_text(self, text: str, metadata: Dict[str, Any] = None) -> List[Dict[st
+            (self._find_paragraph_boundary, "paragraph"),
+            (self._find_sentence_boundary, "sentence"),
+            (self._find_line_boundary, "line"),
+            (self._find_word_boundary, "word")
+        ]
+        
+        for finder, boundary_type in boundary_finders:
+            best_break = finder(search_text, ideal_end)
+            if best_break is not None:
+                adjusted_text = search_text[:best_break].strip()
+                # Convert back to token index using single encode call
+                add_special_tokens = self.tokenizer_type != "qwen"
```

Ruff (0.12.2)

Local variable `add_special_tokens` is assigned to but never used

Remove assignment to unused variable `add_special_tokens`

(F841)